xc/device.ml

```diff
@@ -1976,7 +1969,14 @@ module Backend = struct
           ~timeout ~cancel ();
         pid
 
-      let stop        = Dm_Common.stop
+      let stop ~xs ~qemu_domid domid =
+        let pid = Qemu.pid ~xs domid in
+        Dm_Common.stop ~xs ~qemu_domid domid;
+        match pid with
+        | None -> () (* nothing to do *)
```

Comment:
Here and in `Db.common.stop`, having a `Opt.(>>=)` would have simplified the code